### ISSUE TITLE
`cargo test` frequently causes stack overflow

### ISSUE THREAD

**Issue by A1Liu** [A1Liu/heaparray-rs#2]
I think this is due to the number of abstractions in this crate; often the actual logic being called is nested beneath 3 or more layers of indirection; this results in *wayyy* too many function calls (when at a low optimization level). I think the only legitimate solution is to add `#[inline]` to the vast majority of the methods, and at the very least to all of the really short ones that act as wrappers.

**Comment by A1Liu** [comment]
Problem seems to happen most frequently when using `cargo test --release`. What could be causing this? I don't think I ever allocate a large array onto the stack, so why would this be happening? Is it because of allocations? Maybe I need to rethink how I'm handling the global allocator?

**Comment by A1Liu** [comment]
I tried removing the monitoring in `src/tests/monitor`:

```rust

// Line 29
impl AllocMonitor for TestMonitor {
    fn monitor(&self, layout: Layout, action: AllocAction) {
        // self.global.monitor(layout, action);
        // self.local.monitor(layout, action);
    }
}
```

and immediately the tests were working correctly again, even in release mode. I guess monitoring allocations using the global allocator was expensive. In fact, I removed just the global monitor and everything seemed to be fine! I didn't really need it anyways, so I should genuinely be good now!
